### ISSUE TITLE
Ansible Local Provision fails to install from PPA on Ubuntu 18.04 LTS (Bionic Beaver)

### ISSUE THREAD

**Issue by githubfoam** [hashicorp/vagrant#9796]
I run "ansible_local". I get " Package python-software-properties has no installation candidate" while I "vagrant up". I think this package is obsolete.Is there a way to make "ansible_local" ignore this package?

### Vagrant version
Installed Version: 2.0.4

### Host operating system
Windows 10

### Guest operating system
Ubuntu-18.04

### Expected behavior
Run "ansible_local" provisioner after successfully installing ansible on Ubuntu-18.04

### Actual behavior
trying to install obsolete package

### Steps to reproduce
1.Vagrant up

**Comment by drewcking** [comment]
I ran into this as well, using Chef's bento/ubuntu-18.04 image, though my host is macOs.

It looks like the problem is that Ubuntu 18 has renamed that package to `python3-software-properties`, but [Vagrant's Ubuntu guest provisioner](https://github.com/hashicorp/vagrant/blob/de1c28aad0f92d05364fbfb9015b9b44d8c8ae10/plugins/provisioners/ansible/cap/guest/ubuntu/ansible_install.rb#L22) is quietly installing `python-software-properties` and `software-properties-common`, in order to get the add-apt-repository program so that it can add the `ansible/ansible` PPA.

Ubuntu 18, however, seems to ship with `add-apt-repository` already installed, so it seems like the appropriate fix in Vagrant is to either check to see if that program already exists, and if not, then install the appropriate packages.

In the short term, I worked around this by specifying in my Vagrantfile that I want Ansible to be installed via pip: `ansible.install_mode = "pip"`

**Comment by gildegoma** [comment]
Ansible getting "mainstream" software level, I think we'll now better ditch ths ansible/ansible PPA and rely on the default `ubuntu` repos (similarly to all other supported linux distros). For version selection, the `pip` install mode will be recommended.

**Comment by gildegoma** [comment]
After a few more verifications, I notice that:
* Ansible packaging in default ubuntu LTS repos is still freezed, and therefore the interest to stick to the `ppa` repo to install latest release remains.
* As of Ubuntu 14.04+, only the package `software-properties-common` is required to get [`add-apt-repository`](http://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html#latest-releases-via-apt-ubuntu) operational.
* @drewcking comments are right (we could optimize by first checking whether `add-apt-repository` is already available, and only install package if necessary)

In conclusion: I'll provide a patch for the next Vagrant release

**Comment by gildegoma** [comment]
There is a proposal to fix this issue in #9879.
